pkg/alertmanager/amcfg_test.go

```diff
@@ -0,0 +1,255 @@
+						Name:      "myamc",
+						Namespace: "mynamespace",
+					},
+					Spec: monitoringv1beta1.AlertmanagerConfigSpec{
+						Route: &monitoringv1beta1.Route{
+							Receiver: "test",
+						},
+						Receivers: []monitoringv1beta1.Receiver{{Name: "test"}},
+					},
+				},
+			},
+			expected: `route:
+  receiver: "null"
+  routes:
+  - receiver: mynamespace-myamc-test
+    match:
+      namespace: mynamespace
+    continue: true
+receivers:
+- name: "null"
+- name: mynamespace-myamc-test
+templates: []
+`,
+		},
+		{
+			name:    "base with subroute, simple CR",
+			kclient: fake.NewSimpleClientset(),
+			baseConfig: alertmanagerConfig{
+				Route: &route{
+					Receiver: "null",
+					Routes:   []*route{{Receiver: "null"}},
+				},
+				Receivers: []*receiver{{Name: "null"}},
+			},
+			amConfigs: map[string]*monitoringv1beta1.AlertmanagerConfig{
+				"mynamespace": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "myamc",
+						Namespace: "mynamespace",
+					},
+					Spec: monitoringv1beta1.AlertmanagerConfigSpec{
+						Route: &monitoringv1beta1.Route{
+							Receiver: "test",
+						},
+						Receivers: []monitoringv1beta1.Receiver{{Name: "test"}},
+					},
+				},
+			},
+			expected: `route:
+  receiver: "null"
+  routes:
+  - receiver: mynamespace-myamc-test
+    match:
+      namespace: mynamespace
+    continue: true
+  - receiver: "null"
+receivers:
+- name: "null"
+- name: mynamespace-myamc-test
+templates: []
+`,
+		},
+		{
+			name: "CR with Pagerduty Receiver",
+			kclient: fake.NewSimpleClientset(
+				&corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "am-pd-test-receiver",
+						Namespace: "mynamespace",
+					},
+					Data: map[string][]byte{
+						"routingKey": []byte("1234abc"),
+					},
+				},
+			),
+			baseConfig: alertmanagerConfig{
+				Route: &route{
+					Receiver: "null",
+				},
+				Receivers: []*receiver{{Name: "null"}},
+			},
+			amConfigs: map[string]*monitoringv1beta1.AlertmanagerConfig{
+				"mynamespace": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "myamc",
+						Namespace: "mynamespace",
+					},
+					Spec: monitoringv1beta1.AlertmanagerConfigSpec{
+						Route: &monitoringv1beta1.Route{
+							Receiver: "test",
+						},
+						Receivers: []monitoringv1beta1.Receiver{{
+							Name: "test",
+							PagerDutyConfigs: []monitoringv1beta1.PagerDutyConfig{{
+								RoutingKey: &corev1.SecretKeySelector{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "am-pd-test-receiver",
+									},
+									Key: "routingKey",
+								},
+							}},
+						}},
+					},
+				},
+			},
+			expected: `route:
+  receiver: "null"
+  routes:
+  - receiver: mynamespace-myamc-test
+    match:
+      namespace: mynamespace
+    continue: true
+receivers:
+- name: "null"
+- name: mynamespace-myamc-test
+  pagerduty_configs:
+  - send_resolved: false
+    routing_key: 1234abc
+templates: []
+`,
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
I'm hesitant to do that in this case. Currently it takes 0.030s to run the tests in this file on my machine (consistently over 5 runs), and following the instructions on the Go wiki for running table driven tests in parallel ([here](https://github.com/golang/go/wiki/TableDrivenTests#parallel-testing)), it takes between 0.030s and 0.032s.

Since the tests are already fast (and parallelization may seemingly slow them down in some cases), I think parallelizing them may be just adding complexity in this case. WDYT?